webhooks_test.go

```
@@ -0,0 +1,680 @@
+	client := gitlabtesting.NewTestClient(t, gitlab.WithBaseURL(settings.GitlabInstance))
+
+	client.MockJobs.EXPECT().
+		ListPipelineJobs(83, int64(1), nil).
+		Return([]*gitlab.Job{{ID: 1, Name: settings.JobName}}, &gitlab.Response{}, nil)
+
+	client.MockJobs.EXPECT().
+		RetryJob(83, int64(1)).
+		Return(&gitlab.Job{ID: 1}, &gitlab.Response{}, nil)
+
+	payload := commentPayload("note", fmt.Sprintf("@%s retry", settings.RetryCommand), 1)
+	request := httptest.NewRequest(http.MethodPost, "/webhook", strings.NewReader(payload))
```

golangci-lint (2.12.2)

[error] net/http/httptest.NewRequest must not be called. use net/http/httptest.NewRequestWithContext

(noctx)